pennylane/devices/default_qubit.py

```diff
@@ -581,6 +591,12 @@ def _setup_execution_config(self, execution_config: ExecutionConfig) -> Executio
         """
         updated_values = {}
 
+        updated_values["convert_to_numpy"] = (
+            execution_config.interface.value not in {"jax", "jax-jit"}
+            or execution_config.gradient_method == "adjoint"
```

Comment:
Is `or` correct here? Do we want to convert to numpy as long as the interface is not jax, regardless of the gradient method or the derivative order?

Reply:
We only have sampling logic for jax, so we currently can't sample with autograd, torch, or tf.  So we need to convert to numpy for them.

Adjoint currently does not support jax arrays, though we could update the algorithm to make it jittable. So right now we need to convert to numpy when using adjoint, even if we are using jax.

And we also need to convert to numpy when doing higher order derivatives, even with jax.  Otherwise the caching won't work correctly. We would get leaked tracers and execute lots of duplicate circuits.